sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/pkversion/PartitionLoadBalancerImpl.java

```diff
@@ -119,30 +119,30 @@ private Mono<Void> run(CancellationToken cancellationToken) {
                         if (cancellationToken.isCancellationRequested()) return Mono.empty();
                         return this.partitionController.addOrUpdateLease(lease);
                     })
-                    .then(Mono.just(this)
+                    .then();
+            })
+            .onErrorResume(throwable -> {
+                // "catch all" exception handler to keep the loop going until the user stops the change feed processor
+                logger.warn("Unexpected exception thrown while trying to acquire available leases", throwable);
+                return Mono.empty();
+            })
+            .then(
+                Mono.just(this)
                         .flatMap(value -> {
                             if (cancellationToken.isCancellationRequested()) {
                                 return Mono.empty();
                             }
-
                             Instant stopTimer = Instant.now().plus(this.leaseAcquireInterval);
                             return Mono.just(value)
                                 .delayElement(Duration.ofMillis(100), CosmosSchedulers.COSMOS_PARALLEL)
-                                .repeat( () -> {
+                                .repeat(() -> {
                                     Instant currentTime = Instant.now();
                                     return !cancellationToken.isCancellationRequested() && currentTime.isBefore(stopTimer);
-                                }).last();
+                                })
+                                .then();
```

Comment:
one more here, lol. We have so many `then()` 🤣

Reply:
Discussed offline, will keep it